ims-apple/IMS Apple/Views/AddingProduct/AddingProductView.swift

```diff
@@ -0,0 +1,172 @@
+    @State private var price: String = ""
+    @State private var availableUnits: String = ""
+    @State private var productDetails: String = ""
+    @State private var state: String = ""
+    @State private var selectedState: String = "Disponible"
+    @State private var showingImagePicker = false
+    @State private var avatarItem: PhotosPickerItem?
+    @State private var avatarImage: Image?
+    
+    var body: some View {
+        VStack {
+            HStack {
+                Text("Datos generales")
+                    .font(.headline)
+                
+                Spacer()
+                
+                Button("Guardar Product") { }
+                    .buttonStyle(GradientButtonStyle(imageLeft: Constants.saveProductImage,
+                                                     buttonHeight: Constants.buttonHeight))
+            }
+            .frame(maxWidth: .infinity, alignment: .leading)
+            
+            HStack(spacing: Constants.horizontalSpacing) {
+                productInformationView
+                
+                productDetailsView
+            }
+            
+            Spacer()
+        }
+        .padding()
+        .frame(maxWidth: .infinity)
+        .background(.grayBackground)
+    }
+    
+    // MARK: - Product Information
+    
+    private var productInformationView: some View {
+        VStack {
+            IMSTextField(
+                type: .custom("Nombre del producto"),
+                text: $productName,
+                hasBorder: Constants.hasBorder,
+                maxWidth: .infinity
+            )
+            
+            IMSTextField(
+                type: .custom("Precio"),
+                text: $price,
+                hasBorder: Constants.hasBorder,
+                maxWidth: .infinity
+            )
+            
+            IMSTextField(
+                type: .custom("Unidades disponibles"),
+                text: $availableUnits,
+                hasBorder: Constants.hasBorder,
+                maxWidth: .infinity
+            )
+            
+            Text("Estado")
+                .frame(maxWidth: .infinity, alignment: .leading)
+            
+            Menu {
+                ForEach(["Disponible", "Ocupado", "Agotado"], id: \.self) { item in
+                    Button(item) {
+                        withAnimation {
+                            selectedState = item
+                        }
+                    }
+                }
+            } label: {
+                Text(selectedState)
+            }
+            .menuStyle(.borderlessButton)
+            .padding(.horizontal)
+            .frame(maxWidth: .infinity, minHeight: Constants.minHeight, alignment: .leading)
+            .overlay {
+                RoundedRectangle(cornerRadius: Constants.cornerRadiusSize)
+                    .stroke(.graySecundary)
+            }
+        }
+    }
+    
+    // MARK: - Product Detail
+    
+    private var productDetailsView: some View {
```

Comment:
NIT: the name is plural but it should be singular.

```suggestion
    private var productDetailView: some View {
```